trio/_core/tests/test_run.py

```diff
@@ -823,20 +844,74 @@ def test_cancel_points():
 
     # if a scope is pending, but then gets popped off the stack, then it
     # isn't delivered
-    with _core.open_cancel_scope() as scope:
+    with _core.CancelScope() as scope:
         scope.cancel()
         await _core.cancel_shielded_checkpoint()
     await _core.checkpoint()
     assert not scope.cancelled_caught
 
 
+async def test_cancel_unbound():
+    async def sleep_until_cancelled(scope):
+        with scope, fail_after(1):
+            await sleep_forever()
+
+    # Cancel before entry
+    scope = _core.CancelScope()
+    scope.cancel()
+    async with _core.open_nursery() as nursery:
+        nursery.start_soon(sleep_until_cancelled, scope)
+
+    # Cancel after entry
+    scope = _core.CancelScope()
+    async with _core.open_nursery() as nursery:
+        nursery.start_soon(sleep_until_cancelled, scope)
+        await wait_all_tasks_blocked()
+        scope.cancel()
+
+    # Cancel after exit, then reuse
+    with _core.CancelScope() as scope:
+        await _core.checkpoint()
+    scope.cancel()
+    await _core.checkpoint()
+    assert scope.cancel_called
+    assert not scope.cancelled_caught
+    with scope:
+        await _core.checkpoint()
+    assert scope.cancelled_caught
+    with scope:
+        assert not scope.cancelled_caught
+        await _core.checkpoint()
+    assert scope.cancelled_caught
+
+    # Attempts to reenter throw an error
+    with _core.CancelScope() as scope:
+        with pytest.raises(RuntimeError) as exc_info:
+            with scope:
+                pass  # pragma: no cover
+        assert "may not be entered while it is already active" in str(
+            exc_info.value
+        )
+        async with _core.open_nursery() as nursery:
+
+            async def try_in_other_task():
+                with pytest.raises(RuntimeError) as exc_info:
+                    with scope:
+                        pass  # pragma: no cover
+                assert "while it is already active in another task" in str(
+                    exc_info.value
+                )
```

Comment:
This test feels a bit odd, given that the scope *is* active in the child task that tries to re-enter it (because it inherited it) :-). Maybe it would be clearer to do something like:

```python
async def enter_scope():
    with scope:
        await sleep_forever()

async with trio.open_nursery() as nursery:
    nursery.start_soon(enter_scope)
    await wait_all_tasks_blocked()

    with pytest.raises(...):
        with scope:
            pass  # pragma: no cover

    nursery.cancel_scope.cancel()
```